librarianp.gemspec

```diff
@@ -17,9 +17,9 @@ Gem::Specification.new do |gem|
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ["lib"]
 
-  gem.add_dependency "thor", "~> 0.15"
+  gem.add_dependency "thor", "~> 1.0"
 
-  gem.add_development_dependency "rake"
+  gem.add_development_dependency 'rake', '< 11.0'
```

Comment:
whats the reason for pining to such an old version?

Reply:
Effectively this https://stackoverflow.com/questions/35893584/nomethoderror-undefined-method-last-comment-after-upgrading-to-rake-11

If I upgrade rspec, I hit a variety of test errors. I can address that here and increase the overall scope of these changes.

This change is designed to allow us to be able to use Facter 4 and librarian-puppet in the same project. Facter 4 relies on Thor 1+.

Reply:
We should not introduce change that pins a dependency to old an version with known CVE's.

https://github.com/advisories/GHSA-jppv-gw3r-w3q8